server/src/main/java/org/elasticsearch/common/time/DateFormatter.java

```diff
@@ -102,18 +98,20 @@
      * @param formatters The list of date formatters to be merged together
      * @return           The new date formtter containing the specified date formatters
      */
-    static DateFormatter merge(DateFormatter ... formatters) {
+    static DateFormatter merge(DateFormatter... formatters) {
         return new MergedDateFormatter(formatters);
     }
 
     class MergedDateFormatter implements DateFormatter {
 
         private final String format;
         private final DateFormatter[] formatters;
+        private final DateMathParser[] dateMathParsers;
```

Comment:
Would you please write a brief doc that this is an optimization that means the formatters don't have to create dateMathParsers each time?  It was non-obvious to me.